functests/vrf/vrf_sriov.go

```diff
@@ -0,0 +1,176 @@
+	}
+
+	apiclient := client.New("")
+	sriovclient := sriovtestclient.New("")
+	var nodesList []string
+
+	execute.BeforeAll(func() {
+		if discovery.Enabled() {
+			Skip("Discovery is not supported.")
+		}
+		err := sriovClean.All()
+		Expect(err).ToNot(HaveOccurred())
+		sriov.WaitStable(sriovclient)
+
+		err = namespaces.Create(sriovNamespaces.Test, apiclient)
+		Expect(err).ToNot(HaveOccurred())
+		sriovclient := sriovtestclient.New("")
+		sriovInfos, err := sriovcluster.DiscoverSriov(sriovclient, namespaces.SRIOVOperator)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(sriovInfos).ToNot(BeNil())
+		nodesList, err = nodes.MatchingOptionalSelectorByName(sriovInfos.Nodes)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(nodesList)).To(BeNumerically(">", 0))
+		sriovDevice, err := sriovInfos.FindOneSriovDevice(nodesList[0])
+		Expect(err).ToNot(HaveOccurred())
+		createSriovPolicy(sriovclient, sriovDevice, nodesList[0], 5, resourceNameVrf)
+
+		ipam := `{"type": "static"}`
+		metaPluginRed := fmt.Sprintf(`{"type": "vrf", "vrfname": "%s"}`, VRFRedName)
+		err = createSriovNetwork(sriovclient, sriovDevice, testNetworkRed, sriovNamespaces.Test, namespaces.SRIOVOperator, resourceNameVrf, ipam, metaPluginRed)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() error {
+			netAttDef := &netattdefv1.NetworkAttachmentDefinition{}
+			return sriovclient.Get(context.Background(), runtimeclient.ObjectKey{Name: testNetworkRed, Namespace: sriovNamespaces.Test}, netAttDef)
+		}, 60*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+		metaPluginBlue := fmt.Sprintf(`{"type": "vrf", "vrfname": "%s"}`, VRFBlueName)
+		err = createSriovNetwork(sriovclient, sriovDevice, testNetworkBlue, sriovNamespaces.Test, namespaces.SRIOVOperator, resourceNameVrf, ipam, metaPluginBlue)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() error {
+			netAttDef := &netattdefv1.NetworkAttachmentDefinition{}
+			return sriovclient.Get(context.Background(), runtimeclient.ObjectKey{Name: testNetworkBlue, Namespace: sriovNamespaces.Test}, netAttDef)
+		}, 60*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+	})
+	AfterEach(func() {
+		err := namespaces.CleanPods(sriovNamespaces.Test, apiclient)
+		Expect(err).ToNot(HaveOccurred())
+		//TODO: We need to remove this block and add cleanup in to sriov-network-operator project
+		err = sriovClean.All()
+		Expect(err).ToNot(HaveOccurred())
+		sriov.WaitStable(sriovclient)
+	})
+
+	Context("", func() {
+		// OCP-36303
+		DescribeTable("Integration: SRIOV, IPAM: static, Interfaces: 1, Scheme: 2 Pods 2 VRFs OCP Primary network overlap",
+			func(node string, ipStack string) {
+				defineDesсribeTable(node, ipStack, sriovNamespaces.Test, apiclient, nodesList[0], testNetworkBlue, testNetworkRed)
+			},
+			Entry(describe, sameNode, ipStackIPv4),
+		)
+	})
+})
+
+func createSriovPolicy(sriovclient *sriovtestclient.ClientSet, sriovDevice *sriovv1.InterfaceExt, testNode string, numVfs int, resourceName string) {
```

Comment:
this is the dpdk implementation.
In case of intel it will create a vfio one, which is not good for what we want to do.
Can you just use the one exported by the sriov operator tests?